### PR TITLE
Use new jaeger-lib metricstest

### LINKED ISSUE #1140: jaeger-lib metricstest
## Requirement - what kind of business use case are you trying to solve?
Upgrade jaeger-lib.

## Problem - what in Jaeger blocks you from solving the requirement?
jaeger-lib made a breaking change in its package structure. It uses `metricstest` instead of `testutils` for metric mocks, etc.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Replace references with to `testutils` with `metricstest`.

### PR DESCRIPTION
**Pull request by isaachier**
## Which problem is this PR solving?
- Resolves #1140. 

## Short description of the changes
- Replace old jaeger-lib API with new jaeger-lib API.

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -17,7 +17,9 @@ import:
   subpackages:
   - transport
 - package: github.com/uber/jaeger-lib
-  version: ^1.5.0
+  repo: https://github.com/isaachier/jaeger-lib
```
Comment (black-adder) [inline glide.yaml]: `/isaachier/jaeger-lib`?

Comment (isaachier) [reply glide.yaml]: My bad. Forgot I was using a personal fork.

**Comment by black-adder** [comment]
CI failed with `Failed to set version on github.com/uber/jaeger-lib to 4aa908b951cc7516c61cf001ddca1118d02d243e: Unable to update checked out version: exit status 128`

**Comment by isaachier** [comment]
Interesting... This isn't pressing. I'll fix later.

**Comment by yurishkuro** [comment]
technically we could upgrade to pinned version, but I would rather wait till 2.0 release of jaeger-lib (https://github.com/jaegertracing/jaeger-lib/issues/47)

**Pull request closed without merging**